pkg/resource/deploy/step_generator.go

```diff
@@ -343,15 +419,36 @@ func (sg *stepGenerator) generateSteps(event RegisterResourceEvent) ([]Step, res
 		return nil, res
 	}
 
+	// Generate the aliases for this resource
+	aliases := make(map[resource.URN]struct{}, 0)
+	for _, alias := range goal.Aliases {
+		urn := sg.collapseAliasToUrn(goal, alias)
+		aliases[urn] = struct{}{}
+	}
+	// Now multiply out any aliases our parent had.
+	if goal.Parent != "" {
+		if parentAlias, has := sg.aliases[goal.Parent]; has {
+			aliases[sg.inheritedChildAlias(goal.Type, goal.Name, goal.Parent.Name(), parentAlias)] = struct{}{}
+			for _, alias := range goal.Aliases {
+				childAlias := sg.collapseAliasToUrn(goal, alias)
+				aliasedChildType := childAlias.Type()
+				aliasedChildName := childAlias.Name()
+				inheritedAlias := sg.inheritedChildAlias(aliasedChildType, aliasedChildName, goal.Parent.Name(), parentAlias)
```

Comment:
Naively this seems to me we are preserving the alias fork bomb behavior here that periodically plagues are users. Every parent alias get multiplied out into an entry in aliases, and if you have grandparent and so on, you get this 20 -> 400 -> 8000 behavior easily that leads to slow processing times. Is this accurate or am I missing something? Was perhaps fixing the perf implications out of scope here for now in this PR?

Reply:
Ah!  The difference here is that we only consider the singular alias the parent _actually_ used, if any.  Previously, we'd iterate over _all_ aliases of the parent resource _and_ all aliases of the child, which caused that multiplication.  In this case, we're simply linear over the number of aliases of each resource.